### ISSUE TITLE
Auto update Material Icon in FileSystem

### ISSUE THREAD

**Issue by o-theophilus** [godotengine/godot#59537]
### Godot version

Godot_v3.4.4-stable_win64.exe

### System information

Win 10

### Issue description

The icon preview and name of materials are not in sync

### Steps to reproduce

1. Add a new SpatiaMaterial to FIleSystem
2. double click on it to bring it settings to the inspector

![Capture](https://user-images.githubusercontent.com/31585224/160232507-7f51fabd-efc7-4be9-ada0-f1d61a180af6.PNG)
3. Rename the spatial material
[Issue 1] you will see the name name is not updated in the inspector
![Capture](https://user-images.githubusercontent.com/31585224/160232637-132fc535-5024-4844-93e4-cb7fe2b931d5.PNG)

4. change the color of the Material Albedo in the inspector
[issue 2] you will see the icon does not update in the FileSystem
![Capture](https://user-images.githubusercontent.com/31585224/160232747-c1c1aea6-ea69-4c8b-9b26-eff967a27231.PNG)

**Comment by fire-forge** [comment]
I think this has to do with how resource previews are generated. It happens to images, scenes, etc. as well.

**Comment by Calinou** [comment]
Indeed, I don't think previews can be generated in real-time without incurring a hefty performance loss.
